src/main/java/net/sf/jabref/gui/desktop/os/NativeDesktop.java

```diff
@@ -1,22 +1,71 @@
 package net.sf.jabref.gui.desktop.os;
 
 import java.io.IOException;
+import java.util.Optional;
 
-public interface NativeDesktop {
-    void openFile(String filePath, String fileType) throws IOException;
+import net.sf.jabref.external.ExternalFileType;
+import net.sf.jabref.external.ExternalFileTypes;
+
+public abstract class NativeDesktop {
```

Comment:
I  don't really like having an interface with default methods. I chose the abstract class because from my point of view a class hierarchy better represents the is-a hierarchy in this case. e.g. a OSX(Desktop) is a Native Desktop. An Interface is more a kind of Role-based thing and in this case not the best way.